jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java

```diff
@@ -95,28 +100,54 @@ public Asn1Object UPER_DecodeBsmStream(InputStream ins) {
 
         try {
             if (ins.available() > 0) {
+                logger.debug("DECODER - Message Size = {}", ins.available());
                 coder.decode(ins, bsm);
                 gbsm = OssBsm.genericBsm(bsm);
             }
-        } catch (Exception e) {
-            logger.debug("Error decoding ", e);
+        } catch (DecodeFailedException e) {
+            AbstractData partialDecodedMessage = e.getDecodedData();
+            if (partialDecodedMessage != null) {
+                logger.debug("DECODER - Warning, message only partially decoded: {}", partialDecodedMessage);
```

Comment:
This one should be a logger.error. We do not expect to receive partial messages.